src/simplexinv.jl

```diff
@@ -1,57 +1,62 @@
 export simplexinv
 
-function simplexinv(c, A, b, IB=0, invB=0; max_iter = 4000)
+function simplexinv(c, A, b, 𝔹=0, invB=0; max_iter = 20000)
   m, n = size(A)
   iter = 0
-  if IB == 0 # construct artificial problem
+  λ = Array{Float64,1}(undef, m); d = Array{Float64,1}(undef, m)
+  if 𝔹 == 0 # construct artificial problem
     artificial = true
     signb = sign.(b)
-    A = [A diagm(signb)]
-    Im = speye(m)
-    IB = collect(n+1:n+m) # indexes of basic variables
-    IN = collect(1:n)
-    co = collect(c)
+    A = [A sparse(Diagonal(signb))]
+    𝔹 = collect(n+1:n+m) # indexes of basic variables
+    ℕ = collect(1:n)
+    co = copy(c)
     c = [zeros(n); ones(m)]
-    invB = collect(diagm(signb))
-    r = -A[:,IN]'*signb # artificial relative costs
-    xB = collect(abs.(b)) # solution in current basis
+    invB = sparse(Diagonal(signb))
+    xB = abs.(b) # solution in current basis
   else
     artificial = false
-    Im = speye(m)
-    IN = setdiff(1:n, IB)
-    r = c[IN] - A[:,IN]' * (invB' * c[IB])
+    if (invB == 0) invB = sparse(inv(Matrix(A[:,𝔹]))) end
+    ℕ = setdiff(1:n, 𝔹)
     xB = invB*b
   end
-  q = findfirst(r .< 0) # Bland's Rule
+  getλ!(λ,c,𝔹)
+  λ = invB'*λ # FIX
+  q = getq(c, λ, A, ℕ)
 
   status = :Optimal
 
-  while !(q == 0 || iter >= max_iter)
+  while !(q == nothing || iter >= max_iter) # relative variable changes to directioner >= max_iter)
     iter += 1
-    @assert all(xB .>= 0)
-    d = invB * A[:,IN[q]] # viable direction
-    apfrac = xB ./ d # relative variable changes to direction
-    indpos = find(d .> 0) # variables that decrease in d direction
-    if length(indpos) == 0
-      status = :Unbounded
-      break
+    getAcol!(d,A,ℕ[q])
+    d = invB * d # viable direction
+
+    xq = Inf
+    for k in 1:m # find min xB/d s.t. d .> 0
+      if d[k] >= eps(Float64)
+        dfrac = xB[k]/d[k]
+        if dfrac < xq
+          xq = dfrac
+          p = k
+        end
+      end
+    end
+    if xq == Inf
+      status = :Unbounded; break
     end
-    indxq = indmin(apfrac[indpos])
-    xq = apfrac[indpos[indxq]]
-    @assert xq >= 0
-    @assert xq < Inf
 
-    p = findfirst(apfrac, xq) # Bland's Rule
-    xB -= xq * d; xB[p] = xq # update solution
-    IB[p], IN[q] = IN[q], IB[p] # update indexes
+    subdot!(xB,d,xq) # update solution
+    xB[p] = xq
+    𝔹[p], ℕ[q] = ℕ[q], 𝔹[p] # update indexes
     #update of inverse of B
-    Q = collect(Im)
+    E = one(zeros(m,m))

```

Comment:
This looks awful. You just want the identity? Use `E = Matrix(1.0I, m, m)`. Though I would prefer the fix to stop using this.